src/main/java/net/earthcomputer/clientcommands/command/arguments/DynamicIntegerArgument.java

```diff
@@ -0,0 +1,77 @@
+package net.earthcomputer.clientcommands.command.arguments;
+
+import com.mojang.brigadier.StringReader;
+import com.mojang.brigadier.arguments.ArgumentType;
+import com.mojang.brigadier.context.CommandContext;
+import com.mojang.brigadier.exceptions.CommandSyntaxException;
+import com.mojang.brigadier.suggestion.Suggestions;
+import com.mojang.brigadier.suggestion.SuggestionsBuilder;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.function.Supplier;
+
+public class DynamicIntegerArgument implements ArgumentType<Integer> {
```

Comment:
you don't need this just use CRangeArgument and predicate if the range is valid from the stack's max size, also if you were to need it, use `MinMaxBounds.Ints`